core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -907,7 +983,16 @@ public void workerRegister(long workerId, List<String> storageTiers,
       }
     }
 
-    registerWorkerInternal(workerId);
+    /**
+     * No locking on the {@link MasterWorkerInfo} is required in this method
+     * as it does not update the object fields.
```

Comment:
This is a bit verbose to be included in the call-site.
For this and some others, could you move such internal info to the method comments itself?